Swift/AEPSampleApp/AdIdUtils.swift

```diff
@@ -0,0 +1,93 @@
+/*
+ Copyright 2022 Adobe
+ All Rights Reserved.
+ 
+ NOTICE: Adobe permits you to use, modify, and distribute this file in
+ accordance with the terms of the Adobe license agreement accompanying
+ it.
+ */
+
+import AdSupport
+import AppTrackingTransparency
+import Foundation
+
+class AdIdUtils {
+    
+    /// Provides the `advertisingIdentifier` for the given environment, assuming tracking authorization is provided.
+    /// Use ``requestTrackingAuthorization(callbackHandler:)`` to request authorization.
+    ///
+    /// Simulators may not provide a valid `UUID`, regardless of authorization state; in this case, use the set ad ID flow to test a specific
+    /// ad ID instead. Please see Apple's documentation on `advertisingIdentifier` for more details:
+    /// https://developer.apple.com/documentation/adsupport/asidentifiermanager/1614151-advertisingidentifier
+    ///
+    /// - Returns: The IDFA in `UUID` format; all-zeros if tracking is not authorized or in simulator environment
+    static func getAdvertisingIdentifierForEnvironment() -> UUID {
+        #if targetEnvironment(simulator)
+        print("""
+            Simulator environment detected; an all-zeros UUID will be retrieved even if
+            authorization is provided. Use the set ad ID flow to set a specific ad ID value.
+            """)
+        #endif
+        print("Advertising identifier: \(ASIdentifierManager.shared().advertisingIdentifier)")
+        return ASIdentifierManager.shared().advertisingIdentifier
+    }
+    
+    /// Checks if ad ID tracking authorization is provided, if not returns `false`. Handles both iOS 14+ and <= iOS 13,
+    /// using the appropriate APIs for each case
+    ///
+    /// - Returns: `true` if authorized, `false` for any other state
+    static func isTrackingAuthorized() -> Bool {
+        if #available(iOS 14, *) {
+            print("Tracking authorization status: \(ATTrackingManager.trackingAuthorizationStatus)")
+            return ATTrackingManager.trackingAuthorizationStatus == .authorized
+        } else {
+            print("iOS version <= 13 detected; using ASIdentifierManager and getting IDFA directly.")
```

Comment:
should this be ` version <= 14 detected`

Reply:
Updated wording according to suggestion for clarity!